### PR TITLE
Fixed #28006 -- Allowed D to be used with lookups on Distance annotation.

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://code.djangoproject.com/ticket/28006

### CODE REVIEW

File: tests/gis_tests/distapp/tests.py [hunk 1]
```diff
@@ -368,6 +368,22 @@ def test_distance_function_raw_result(self):
...
+        ).filter(d=D(m=1))
+        self.assertTrue(qs.exists())
+
+    @skipIfDBFeature("supports_distance_geodetic")
+    @skipUnlessDBFeature("has_Distance_function")
+    def test_distance_function_raw_result_d_lookup(self):
+        qs = Interstate.objects.annotate(
+            d=Distance(Point(0, 0, srid=4326), Point(0, 1, srid=4326)),
+        ).filter(d=D(m=1))
+        with self.assertRaisesMessage(ValueError, 'Distance measure is supplied, but units are unknown for result.'):
```
Comment (timgraham) [inline tests/gis_tests/distapp/tests.py]: I'd use the `msg = '...'` style here to shorten this line.

**Comment by timgraham** [comment]
buildbot, test on oracle.

**Comment by sir-sigurd** [comment]
buildbot, test on oracle.

**Comment by sir-sigurd** [comment]
buildbot, test on oracle.

**Comment by sir-sigurd** [comment]
Oracle failures are fixed, I checked it locally. Buildbot doesn't want to test on Oracle for me :-(.

**Comment by sir-sigurd** [comment]
buildbot, test on oracle.

**Comment by sir-sigurd** [comment]
Oracle failures are fixed now.

**Pull request merged**